crates/bevy_scene/src/scene_filter.rs

```diff
@@ -0,0 +1,222 @@
+use bevy_utils::hashbrown::hash_set::IntoIter;
+use bevy_utils::HashSet;
+use std::any::{Any, TypeId};
+
+/// A filter used to control which types can be added to a [`DynamicScene`].
+///
+/// This scene filter _can_ be used more generically to represent a filter for any given type;
+/// however, note that its intended usage with `DynamicScene` only considers [components].
+/// Adding types that are not components will have no effect when used with `DynamicScene`.
+///
+/// [`DynamicScene`]: crate::DynamicScene
+/// [components]: bevy_ecs::prelude::Component
+#[derive(Default, Debug, Clone, PartialEq, Eq)]
```

Comment:
is this `PartialEq` implementation correct? should `SceneFilter::None` be equivalent to `SceneFilter::Allowlist` with no elements?

Reply:
I don't think so. It should be based on variant and `HashSet` imo rather than the semantics of their values. That would cause the least amount of confusion (since the cross-variant equality only applies to `None` and an empty `Denylist`)